thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/function/WeekOverWeekRuleFunction.java

```diff
@@ -157,27 +150,8 @@ public static String parseWowString(String wowString) {
     return wowString.substring(head, tail);
   }
 
-  @Override public DataModel getDataModel() {
-    return dataModel;
-  }
-
-  @Override public List<TransformationFunction> getCurrentTimeSeriesTransformationChain() {
-    return currentTimeSeriesTransformationChain;
-  }
-
-  @Override public List<TransformationFunction> getBaselineTimeSeriesTransformationChain() {
-    return baselineTimeSeriesTransformationChain;
-  }
-
-  @Override public PredictionModel getPredictionModel() {
-    return predictionModel;
-  }
-
-  @Override public DetectionModel getDetectionModel() {
-    return detectionModel;
-  }
-
-  @Override public MergeModel getMergeModel() {
-    return mergeModel;
+  @Override
+  public boolean useHistoryAnomaly() {
+    return false;
```

Comment:
Is it a conflict that if one anomaly function defines "AnomalyRemoval" functionality in  Transformation chain, but useHistoryAnomaly() is false. Or if use the new default value of "useHistoryAnomaly()" to be true, but failed to define any "AnomalyRemoval" strategy. Shall we come up with a way to finally unify "useHistoryAnomaly()" and "AnomalyRemoval()" in transformation chain?

Reply:
This is a good question. We can examine the transformation functions and figure out if useHitoryAnomaly() should be true. But it can lead to an ad-hoc design that if there is new model requiring history anomalies, we don't know that. We can discuss a more systematic solution for this.